components/ActivityForm/index.js

```diff
@@ -2,16 +2,71 @@ import styled from "styled-components";
 import { StyledButton } from "@/components/StyledButton";
 import { useRouter } from "next/router";
 import ImageUploading from "react-images-uploading";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import Image from "next/image";
 import { Icon } from "@/components/Icon";
+import Select from "react-select";
+import {
+  fetchCitiesData,
+  fetchCoordinatesData,
+  fetchCountriesData,
+} from "@/pages/api/geoData";
+import dynamic from "next/dynamic";
+
+const MapComponent = dynamic(() => import("@/components/Map"), { ssr: false });
 
 export default function ActivityForm({ onSubmit, initialData, isEditMode }) {
   const router = useRouter();
   const defaultImages = initialData ? initialData.images : [];
   const [images, setImages] = useState(defaultImages);
   const maxNumberOfImages = 20;
 
+  const [countries, setCountries] = useState([]);
+  const [cities, setCities] = useState([]);
+  const [selectedCountry, setSelectedCountry] = useState(
+    initialData
+      ? { value: initialData?.country, label: initialData?.country }
+      : null
+  );
+  const [selectedCity, setSelectedCity] = useState(
+    initialData ? { value: initialData?.city, label: initialData?.city } : null
+  );
+  const [coordinates, setCoordinates] = useState(
+    initialData
+      ? { lat: initialData.lat, lng: initialData.lng }
+      : { lat: 0, lng: 0 }
+  );
+
+  useEffect(() => {
+    async function fetchCountries() {
+      const countriesNames = await fetchCountriesData();
+      setCountries(countriesNames);
+    }
+    fetchCountries();
+  }, []);
```

Comment:
just an idea: You could save the result of the fetch and provide it with your app as initial data. So the countries are always available from the start and you could save one fetch. Pro: Better performance. Con: New States are not automatically included.